File: quote/Projects/SampleProperties/ComponentPropertiesFactory.cs
﻿using System;

using Model.Common;
using Host;

namespace SampleProperties
{
    [Register(Key = typeof(Model.Template.Ext.IComponentPropertiesFactory))]
    public class ComponentPropertiesFactory : Model.Template.Ext.IComponentPropertiesFactory
    {
        public Model.Common.ComponentProperties CreateComponentProperties(Model.Template.Detail detail)
        {
            return new SampleComponentProperties(detail);
        }
    }
}

File: quote/Projects/DekalbProperties/ComponentPropertiesFactory.vb
Imports Host

<Register(Key:=GetType(Model.Template.Ext.IComponentPropertiesFactory))>
Public Class ComponentPropertiesFactory
    Implements Model.Template.Ext.IComponentPropertiesFactory

Public Function CreateComponentProperties(detail As Model.Template.Detail) As Model.Common.ComponentProperties _ 
    Implements Model.Template.Ext.IComponentPropertiesFactory.CreateComponentProperties
        Return New DisplayableComponentProperties(New DekalbComponentProperties(detail))
    End Function

End Class
